javascript/readme.md

```diff
@@ -1,3 +1,386 @@
+	- === and !== Operators.
+	- [Confusing Pluses and Minuses](#confusing-pluses-and-minuses)
+	- [eval is Evil](#eval-is-evil)
+
+## JavaScript Files
+
+JavaScript programs should be stored in and delivered as .js files.
+
+JavaScript code should not be embedded in HTML files unless the code is specific to a single session. Code in HTML adds significantly to pageweight with no opportunity for mitigation by caching and compression.
+
+## Whitespace
+
+Blank spaces should always be used in the following circumstances:
+
+- A keyword followed by `(` left parenthesis should be separated by a space. Spaces are used to make things that are not invocations look less like invocations, so for example, there should be space after `if` or `while`.
+- A blank space should not be used between a function value and its invoking `(` left parenthesis. This helps to distinguish between keywords and function invocations.
+- The word function is always followed with one space.
+- No space should separate a unary operator and its operand except when the operator is a word such as typeof.
+- All binary operators should be separated from their operands by a space on each side except `.` period and `(` left parenthesis and `[` left bracket.
+- Every `,` comma should be followed by a space or a line break.
+- Each `;` semicolon at the end of a statement should be followed with a line break.
+- Each `;` semicolon in the control part of a for statement should be followed with a space.
+
+Every statement should begin aligned with the current indentation. The outermost level is at the left margin. The indentation increases by 4 spaces when the last token on the previous line is { left brace, [ left bracket, ( left paren. The matching closing token will be the first token on a line, restoring the previous indentation.
+
+The ternary operator can be visually confusing, so ? question mark always begins a line and increase the indentation by 4 spaces, and : colon always begins a line, aligned with the ? question mark.
+
+```javascript
+var integer = function (
+    value,
+    default_value
+) {
+    value = resolve(value);
+    return (typeof value === "number")
+        ? Math.floor(value)
+        : (typeof value === "string")
+            ? value.charCodeAt(0)
+            : default_value;
+};
+```
+
+If `.` period is the first character on a line, the indentation is increased by 4 spaces.
+
+Avoid excessively long lines. When a statement will not fit nicely on a single line, it may be necessary to break it. It is best to break after a `{` left brace, `[` left bracket, `(` left paren, `,` comma, or before a `.` period, `?` question mark, or `:` colon. If such a break is not feasible, then break after an operator and continue on the next line with 8 spaces added to the current indentation. Those 8 spaces do not change the current indentation.
+
+Clauses (`case`, `catch`, `default`, `else`, `finally`) are not statements and so should not be indented like statements.
```

Comment:
Does this mean you don't allign the 'else' with the initial if statement? Maybe some examples for this would be good, especially how you see a switch statement with 'case' and 'default' working.